test/e2e/performanceprofile/functests/1_performance/irqbalance.go

```diff
@@ -0,0 +1,137 @@
+			mcps.WaitForCondition(mcpName, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+		}
+	})
+
+	Context("Verify IRQ Balancing", func() {
+
+		var testpod *corev1.Pod
+
+		AfterEach(func() {
+			deleteTestPod(testpod)
+		})
+
+		It("Should not overwrite the banned CPU set on tuned restart", func() {
+			if profile.Status.RuntimeClass == nil {
+				Skip("runtime class not generated")
+			}
+			runtimeClass := profile.Status.RuntimeClass
+
+			irqLoadBalancingDisabled := profile.Spec.GloballyDisableIrqLoadBalancing != nil && *profile.Spec.GloballyDisableIrqLoadBalancing
+			if irqLoadBalancingDisabled {
+				Skip("this test needs dynamic IRQ balancing")
+			}
+
+			node := workerRTNodes[0] // any random node is fine
+			By(fmt.Sprintf("verifying worker node %q", node.Name))
+
+			var err error
+
+			bannedCPUs, err := nodes.BannedCPUs(node)
+			Expect(err).ToNot(HaveOccurred(), "failed to extract the banned CPUs from node %q", node.Name)
+			Expect(bannedCPUs.IsEmpty()).To(BeTrue(), "banned CPUs %v not empty on node %q", bannedCPUs, node.Name)
+
+			smpAffinitySet, err := nodes.GetDefaultSmpAffinitySet(&node)
+			Expect(err).ToNot(HaveOccurred(), "failed to get default smp affinity")
+
+			onlineCPUsSet, err := nodes.GetOnlineCPUsSet(&node)
+			Expect(err).ToNot(HaveOccurred(), "failed to get Online CPUs list")
+
+			Expect(smpAffinitySet.Equals(onlineCPUsSet)).To(BeTrue(), "found default_smp_affinity %v, expected %v", smpAffinitySet, onlineCPUsSet)
+
+			cpuRequest := 2 // minimum amount to be reasonably sure we're SMT-aligned
+			testpod = getStressPod(node.Name, cpuRequest)
+			testpod.Namespace = testutils.NamespaceTesting
+			testpod.Annotations = map[string]string{
+				"irq-load-balancing.crio.io": "disable",
+				"cpu-quota.crio.io":          "disable",
+			}
+			testpod.Spec.NodeName = node.Name
+			testpod.Spec.RuntimeClassName = runtimeClass
+			promotePodToGuaranteed(testpod)
+
+			err = testclient.Client.Create(context.TODO(), testpod)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = pods.WaitForCondition(testpod, corev1.PodReady, corev1.ConditionTrue, 10*time.Minute)
+			logEventsForPod(testpod)
+			Expect(err).ToNot(HaveOccurred())
+
+			// now we have something in the IRQBalance cpu list. Let's make sure the restart doesn't overwrite this data.
+			postCreateBannedCPUs, err := nodes.BannedCPUs(node)
+			Expect(err).ToNot(HaveOccurred(), "failed to extract the banned CPUs from node %q", node.Name)
+			Expect(postCreateBannedCPUs.IsEmpty()).To(BeFalse(), "banned CPUs %v not empty on node %q", postCreateBannedCPUs, node.Name)
+
+			By(fmt.Sprintf("getting a TuneD Pod running on node %s", node.Name))
+			pod, err := util.GetTunedForNode(cs, &node)
+			Expect(err).NotTo(HaveOccurred())
+
+			By(fmt.Sprintf("causing a restart of the tuned pod (deleting the pod) on %s", node.Name))
+			_, _, err = util.ExecAndLogCommand("oc", "delete", "pod", "-n", pod.Namespace, pod.Name)
```

Comment:
So, these days the `oc/kubectl delete` commands block until a resource is deleted unless one says `--wait=false`.  However, this still makes me a bit uneasy because it didn't use to be that way in the past, in which case we'd have a race here.  Do we want/need to say `--wait=true` to make it more obvious/explicit there is no race here?

Reply:
sure thing (will update both PRs)